pypeit/scripts/show_2dspec.py

```diff
@@ -48,25 +45,37 @@ def show_trace(specobjs, det, viewer, ch):
         viewer (?):
         ch (?):
     """
-    if specobjs is None:
+    if sobjs is None:
         return
-    in_det = np.where(specobjs.DET == det)[0]
+    in_det = np.where(sobjs.DET == det)[0]
+    trace_list = []
+    trc_name_list = []
+    maskdef_extr_list = []
+    manual_extr_list = []
     for kk in in_det:
-        trace = specobjs[kk]['TRACE_SPAT']
-        obj_id = specobjs[kk].NAME
-        maskdef_objname = specobjs[kk].MASKDEF_OBJNAME
-        maskdef_extr_flag = specobjs[kk].MASKDEF_EXTRACT
-        manual_extr_flag = specobjs[kk].hand_extract_flag
+        trace = sobjs[kk]['TRACE_SPAT']
+        obj_id = sobjs[kk].NAME
+        maskdef_objname = sobjs[kk].MASKDEF_OBJNAME
+        maskdef_extr_flag = sobjs[kk].MASKDEF_EXTRACT
+        manual_extr_flag = sobjs[kk].hand_extract_flag
         if maskdef_objname is not None:
```

Comment:
trc_name = '{} OBJNAME:{}'.format(obj_id, maskdef_objname) if maskdef_objname is None else obj_id